Client/game_sa/CBuildingsPoolSA.cpp

```diff
@@ -372,7 +373,8 @@ void CBuildingsPoolSA::RemovePedsContactEnityLinks()
             ped->pLastContactedEntity[2] = nullptr;
             ped->pLastContactedEntity[3] = nullptr;
             ped->m_ucCollisionState = 0;
-            ped->pTargetedEntity = nullptr;
```

Comment:
This line exists to prevent crashes

Reply:
if pTragetEntity is lastDamagedEntity now, you should replace it here too.

Reply:
Look at your refactor PR closelly.
You renamed pTargetedEntity to pLookAtEntity there and added a new field with name pTargetedEntity

Reply:
What a mess here.
So, after refactor it should be `ped->pLookAtEntity = nullptr`

Reply:
And your GetTargetedEntity should reuse this field, to restore old behavior.
Now it uses something related to mouse.

Reply:
Old CPedSAInterface
![image](https://github.com/user-attachments/assets/04c01939-0de7-4da7-9432-dad0e02a4f41)

New CPedSAInterface
![image](https://github.com/user-attachments/assets/e37ffc3c-9739-4169-9061-adba17b35915)

New CPlayerPedSA
![image](https://github.com/user-attachments/assets/dd81f00e-5145-4e5b-956d-4354f69ed82d)

The field named ``pTargetedEntity`` has always pointed to the same offset 🤔

Reply:
ah, ok, i didn't notice pad[104] there, sorry